src/main/java/io/divolte/server/BrowserEventData.java

```diff
@@ -41,6 +44,7 @@
     public final Optional<Integer> screenPixelHeight;
     public final Optional<Integer> devicePixelRatio;
     public final Function<String, Optional<String>> eventParameterProducer;
+    public final Supplier<Map<String,String>> eventParametersProducer;
 
```

Comment:
It does add some overhead when grabbing only a single param. If you're not
mapping the entire map, there's no need to materialise it.

On Fri, Feb 13, 2015 at 12:01 PM, Andrew Snare <notifications@github.com>
wrote:

> In src/main/java/io/divolte/server/BrowserEventData.java
> <https://github.com/divolte/divolte-collector/pull/79#discussion_r24657181>
> :
>
> > @@ -41,6 +44,7 @@
> >      public final Optional<Integer> screenPixelHeight;
> >      public final Optional<Integer> devicePixelRatio;
> >      public final Function<String, Optional<String>> eventParameterProducer;
> > +    public final Supplier<Map<String,String>> eventParametersProducer;
> >
>
> Do we still need to keep the eventParameterProducer field? It's kind of
> redundant now that we have a producer for all of the parameters.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/divolte/divolte-collector/pull/79/files#r24657181>.
>